paper-ripple.html

```diff
@@ -116,362 +113,418 @@
-      var dy = y - (ctx.containerHeight / 2);
+    function Ripple (element) {
+      this.element = element;
+      this.color = window.getComputedStyle(element).color;
 
-      ctx.wc.style.webkitTransform = 'translate3d(' + dx + 'px,' + dy + 'px,0)';
-      ctx.wc.style.transform = 'translate3d(' + dx + 'px,' + dy + 'px,0)';
+      this.wave = document.createElement('div');
+      this.waveContainer = document.createElement('div');
+      this.wave.style.backgroundColor = this.color;
+      this.wave.classList.add('wave');
+      this.waveContainer.classList.add('wave-container');
+      this.waveContainer.appendChild(this.wave);
+      this.container.appendChild(this.waveContainer);
+      this.background.style.backgroundColor = this.color;
 
-      // 2d transform for safari because of border-radius and overflow:hidden clipping bug.
-      // https://bugs.webkit.org/show_bug.cgi?id=98538
-      ctx.wave.style.webkitTransform = 'scale(' + s + ',' + s + ')';
-      ctx.wave.style.transform = 'scale3d(' + s + ',' + s + ',1)';
+      this.resetInteractionState();
     }
 
-    //
-    // SETUP
-    //
-    function createWave(elem) {
-      var elementStyle = window.getComputedStyle(elem);
-      var fgColor = elementStyle.color;
-
-      var inner = document.createElement('div');
-      inner.style.backgroundColor = fgColor;
-      inner.classList.add('wave');
-
-      var outer = document.createElement('div');
-      outer.classList.add('wave-container');
-      outer.appendChild(inner);
-
-      var container = elem.$.waves;
-      container.appendChild(outer);
-
-      elem.$.bg.style.backgroundColor = fgColor;
-
-      var wave = {
-        bg: elem.$.bg,
-        wc: outer,
-        wave: inner,
-        waveColor: fgColor,
-        maxRadius: 0,
-        isMouseDown: false,
-        mouseDownStart: 0.0,
-        mouseUpStart: 0.0,
-        tDown: 0,
-        tUp: 0
-      };
-      return wave;
-    }
+    Ripple.MAX_RADIUS = 150;
 
-    function removeWaveFromScope(scope, wave) {
-      if (scope.waves) {
-        var pos = scope.waves.indexOf(wave);
-        scope.waves.splice(pos, 1);
-        // FIXME cache nodes
-        wave.wc.remove();
-      }
-    };
+    Ripple.prototype = {
+      get container () {
+        return this.element.$.waves;
+      },
 
-    // Shortcuts.
-    var pow = Math.pow;
-    var now = Date.now;
-    if (window.performance && performance.now) {
-      now = performance.now.bind(performance);
-    }
+      get background () {
+        return this.element.$.background;
+      },
 
-    function cssColorWithAlpha(cssColor, alpha) {
-        var parts = cssColor.match(/^rgb\((\d+),\s*(\d+),\s*(\d+)\)$/);
-        if (typeof alpha == 'undefined') {
-            alpha = 1;
-        }
-        if (!parts) {
-          return 'rgba(255, 255, 255, ' + alpha + ')';
+      get recenters () {
+        return this.element.classList.contains('recenteringTouch');
+      },
+
+      get mouseDownElapsed () {
```

Comment:
I would describe it as a preference for getters vs. temporary variables. Although I admit that I did replace one or two free-floating functions with getters.

When a routine does not need to be parameterized, and its behavioral scope requires state that the class should already encapsulate, I will intuitively prefer a getter.

Do you prefer methods? If so, why? Is there a rule we can codify in the style guide?